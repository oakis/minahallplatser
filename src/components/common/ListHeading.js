import React from 'react';
import { View } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import { Text, Spinner } from './';
import { component, metrics, colors } from '../style';

export const ListHeading = ({ text, icon, iconSize = 20, onPress, loading }) => {
    return (
        <View style={{ flex: 1, flexDirection: 'row', justifyContent: 'center', alignItems: 'center', height: 29 }}>
            <Text style={[component.text.heading, { height: 29 }]}>{text}</Text>
            <View style={{ flex: 1, marginTop: metrics.margin.md, paddingRight: metrics.padding.md, borderBottomWidth: 2, borderColor: colors.primary, height: 29, justifyContent: 'center' }}>
                {loading ? <Spinner color={colors.primary} noFlex left />
                : <Icon name={icon} onPress={onPress} size={iconSize} style={{ textAlign: 'left' }} />}
            </View>
        </View>
    );
};
