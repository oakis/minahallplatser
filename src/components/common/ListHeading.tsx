import React from 'react';
import {View} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {Text, Spinner} from '@common';
import {component, metrics, colors} from '@style';

type ListHeadingProps = {
  text: string;
  icon: string;
  iconSize: number;
  onPress: () => void;
  loading: boolean;
  style: object;
};

export const ListHeading = ({
  text,
  icon,
  iconSize = 24,
  onPress,
  loading = false,
  style = {},
}: ListHeadingProps): JSX.Element => {
  return (
    <View>
      <View
        style={[
          {
            flexDirection: 'row',
            justifyContent: 'center',
            alignItems: 'center',
            height: 29,
          },
          style,
        ]}>
        <Text style={[component.text.heading, {height: 29}]}>{text}</Text>
        <View
          style={{
            flex: 1,
            marginTop: metrics.margin.md,
            paddingRight: metrics.padding.md,
            height: 29,
            justifyContent: 'center',
          }}>
          {loading ? (
            <Spinner color={colors.primary} right />
          ) : (
            <Icon
              name={icon}
              onPress={onPress}
              size={iconSize}
              style={{textAlign: 'right'}}
            />
          )}
        </View>
      </View>
      <View
        style={{height: 2, width: '100%', backgroundColor: colors.primary}}
      />
    </View>
  );
};
