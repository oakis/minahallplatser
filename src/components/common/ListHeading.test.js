import React from 'react';
import { shallow } from 'enzyme';
import { ListHeading } from './ListHeading';

it('should match snapshot', () => {
    const wrapper = shallow(<ListHeading />);
    expect(wrapper).toMatchSnapshot();
});
