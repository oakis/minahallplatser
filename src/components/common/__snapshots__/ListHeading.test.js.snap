// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 29,
          "justifyContent": "center",
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginTop": 10,
            "paddingLeft": 10,
            "paddingRight": 5,
          },
          Object {
            "height": 29,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "height": 29,
          "justifyContent": "center",
          "marginTop": 10,
          "paddingRight": 10,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#779ECB",
        "height": 2,
        "width": "100%",
      }
    }
  />
</View>
`;
